flake.nix

```diff
@@ -11,31 +11,44 @@
           pkgs = import nixpkgs {
             inherit system overlays;
           };
+          gohip-vendor-hash = import ./gohip.vendor.hash.nix;
           gohip-package = (pkgs.buildGoModule {
             name = "gohip";
             pname = "gohip";
 
             src = ./.;
-            vendorHash = "sha256-jZcR+OMiGplA5yVMpQT4qNKH+tyFp7PXQVoG+oghBLs=";
+            vendorHash = gohip-vendor-hash pkgs;
             proxyVendor = true;
             excludedPackages = ["osdata" "others" "systemd" "types"];
           });
 
+          upgrade-nix-gohip-package = pkgs.writeShellScriptBin "upgrade-nix-gohip" ''
+            echo "UPGRADING"
+            echo "pkgs : pkgs.lib.fakeHash" > $WORKSPACE/gohip.vendor.hash.nix
+            NEW_HASH=`nix build .#default 2>&1 | grep " got: " | awk '{print $2}'`
+            echo "pkgs : \"$NEW_HASH\"" > $WORKSPACE/gohip.vendor.hash.nix
```

Comment:
the inner quotes are needed ? (naive question, i dont know nix)

Reply:
Yes, they are needed. This line generates a very minor nix module which is a lambda function (a la Haskell), that says given a `pkgs` input variable, return a string. The file is committed, so you can take a look [here](https://github.com/bechampion/gohip/blob/fd75cde93c788f5b8df2b34dca3c0a7779b60822/gohip.vendor.hash.nix) to see how it would look like